packages/test/test-end-to-end-tests/src/test/container.spec.ts

```diff
@@ -137,6 +140,26 @@ describeNoCompat("Container", (getTestObjectProvider) => {
 			"Client details should be set with interactive as true",
 		);
 	});
+	it.only("Delta manager receives pong event", async () => {
+		const container = await createConnectedContainer();
+		const realSetTimeout = setTimeout;
+		// function to yield control in the Javascript event loop.
+		async function yieldEventLoop(): Promise<void> {
+			await new Promise<void>((resolve) => {
+				realSetTimeout(resolve, 0);
+			});
+		}
+
+		// await requestFluidObject<ITestDataObject>(container, "default");
+		let run = 0;
+		container.deltaManager.on("pong", () => {
+			run++;
+		});
+
+		clock.tick(60000);
```

Comment:
60 seconds might not be enough - your test might fail if ping/pong takes some time (unless we assume here that ping/pong process starts right on connection, and thus these 60 seconds are supposed to cover actual ping roundtrip time, not time it takes to schedule next iteration that will trigger ping).

I think fake timers help here as we register for pong automatically (in connection manager), so we might miss first pong, and faking timers helps us get to it faster. But it may take real amount of time (unknown) to get actual pong going over the network, so any amount of time you put here (given its fake time might not be enough.

Can the test be structured in a way where
- we move fake time forward by 60 seconds.
- after that we wait for pong callback and exit right away when we receive it?

That way it would work Ok no matter how long it takes (subject to overall test timeouts).
You might want to increase mocha test timeout here though.